src/DSharp.Mscorlib/Array.cs

```diff
@@ -33,35 +37,48 @@ public sealed class Array : ICollection
 
         public extern Array Filter(ArrayItemFilterCallback itemFilterCallback);
 
+        // I guess these two fro each methods are just js forEach.
+        // ForEach in CLR looks like this -> static void ForEach<T>(T[] array, Action<T> action)
         public extern void ForEach(ArrayCallback callback);
 
         public extern void ForEach(ArrayItemCallback itemCallback);
 
         public extern IEnumerator GetEnumerator();
 
+        // No GetRange methods in CLR
         public extern Array GetRange(int index);
 
         public extern Array GetRange(int index, int count);
 
+        // IndexOf is always static in CLR
+        // The most similar in CLR is -> static int IndexOf(Array array, object? value)
         public extern int IndexOf(object item);
 
+        // The most similar in CLR is -> static int IndexOf(Array array, object? value, int startIndex)
```

Comment:
There is a magic method in the compiler: `int IndexOf<T>(T value)`, this is what is called on you're arrays of `T[] arr = new t[0]`
We can deprecate this api